ddl/placement/constraints.go

```diff
@@ -110,3 +115,34 @@ func (constraints *Constraints) Add(label Constraint) error {
 	}
 	return nil
 }
+
+// FingerPrint returns a unique string for the constraints.
+func (constraints Constraints) FingerPrint() string {
```

Comment:
`Constraints` => `*Constraints`? I got a warning in IDE: "Struct Constraints has methods on both value and pointer receivers. Such usage is not recommended by the Go Documentation. "